Feature: 프로필 페이지 구현

Monixc opened the pull request:
### 📌 PR 종류 (PR Type)
<!-- 아래 항목 중 해당하는 곳에 [x] 로 표시해주세요. -->

- [ ] 🐛 버그 수정 (Bugfix)
- [x]  ✨ 기능 추가 (Feature)
- [x] 🎨 UI 스타일 변경 (Style) - CSS, 클래스명, 레이아웃 등 UI 변경
- [ ] ♻️ 코드 리팩토링 (Refactor) - 기능 변화 없이 코드 구조 개선
- [ ] 📦 패키지/빌드 관련 변경 (Build) - 설정 파일 수정, 의존성 추가 등
- [ ] 🧱 테스트 코드 추가/수정 (Test)
- [ ] 📝 문서 수정 (Docs)
- [ ] 🚧 WIP (작업 중 PR)
- [ ] 기타 (Other): <!-- 직접 입력 -->

---

### 📝 요약 (Summary)
<!-- 한두 문장으로 변경 내용을 요약해주세요. -->
프로필 페이지(my, developer) 페이지 퍼블리싱 및 라우팅 경로 추가

---

### 🔍 상세 내용 (Describe your changes)
<!-- 어떤 변경사항이 있었는지, 왜 변경했는지 상세히 작성해주세요. 코드 흐름, 주요 로직 등을 설명해도 좋습니다. -->
ex)
프로필 페이지 퍼블리싱
- 이미지 영역 클릭 시 파일 업로드 화면

개발자 프로필 페이지 퍼블리싱
- 프로필, 작성한 글 탭 선택에 따라 해당하는 컴포넌트 렌더링 도도록 구성


---



### 📸 스크린샷 (필수)
<!-- UI 변경이 있는 경우, 변경 전/후 비교나 주요 화면을 첨부해주세요. -->
<img width="826" height="1268" alt="image" src="https://github.com/user-attachments/assets/344fb53f-4a90-46bd-b151-2b788b647c12" />
<img width="815" height="1275" alt="image" src="https://github.com/user-attachments/assets/7b70121e-bfd5-4ae6-8fb1-52261144ba77" />


---

### ✅ 확인 사항 (CheckList)
<!-- PR을 보내기 전 스스로 점검할 수 있도록 합니다. 팀원이 리뷰할 때도 참고할 수 있습니다. -->
- [x] 브라우저에서 정상 동작 확인
- [x] 기존 기능에 영향 없음
- [x] 반응형 UI 대응 확인
- [x] 코드 컨벤션에 맞게 작성됨 (eslint, prettier 등)
- [ ] 테스트 코드 추가/수정 또는 테스트 통과 확인

---
### 💬 기타 참고사항 (Optional)
<!-- 리뷰어가 알면 좋을 추가 정보, 요청 사항 등을 작성해주세요. -->

Monixc committed:
feat: my, developer 라우팅 경로 추가

Monixc committed:
feat: my, developer 페이지 퍼블리싱

Monixc committed:
Merge branch 'develop' of https://github.com/prgrms-fullstack-devcourse/Comeet_FE into feature/page-profile

Monixc committed:
feat: 개발자 페이지 탭 및 프로필, 작성글 컴포넌트 추가

- App.tsx에서 온보딩 페이지 관련 코드 제거 및 GlobalLayout로 변경
- profile.ts에 개발자 탭 상수 추가
- DeveloperPage에서 AppTabs 컴포넌트로 탭 기능 구현
- 프로필 및 작성글 컴포넌트 추가

Monixc committed:
fix: DeveloperPage 레이아웃 수정

- ProfileSection을 px-4로 감싸 여백 추가
- 텍스트 색상 및 여백 조정으로 가독성 향상

Monixc committed:
fix: ProfileSection 이미지 미리보기 로직 개선

- FileReader의 결과가 문자열인지 확인하여 setImagePreview 호출 시 타입 안전성 강화

Monixc committed:
fix: DeveloperPage 탭 변경 로직 개선

- 유효한 탭 값인지 확인하여 setActiveTab 호출 시 타입 안전성 강화

Monixc committed:
fix: 마이페이지 메뉴 관련 constants 타입 강화

ksy90101 commented on src/components/common/ProfileSection.tsx:
```diff
@@ -0,0 +1,71 @@
+          )}
+        </div>
+        {isEditable && (
+          <input
+            type="file"
+            ref={fileInputRef}
+            onChange={handleFileChange}
+            className="hidden"
+            accept="image/*"
+          />
```
file input과 같은 컴포넌트는 재사용성이 강합니다.
따라서 따로 컴포넌트 분리를 통해 추상화를 해보시면 좋겠습니다.

ksy90101 commented on src/components/common/ProfileSection.tsx:
이 컴포넌트가 common인 근거가 무엇인가요?

Monixc replied:
사용자 프로필 이미지 표시 컴포넌트로, my 페이지와 developer 페이지에서 사용합니다.
두 페이지는 각각 본인 프로필 표시, 타인 프로필 표시로 구분하여 구성하였고, 특정 페이지에 종속되지 않고 두 페이지에 걸쳐 사용하는 컴포넌트여서 common으로 구성했습니다.

ksy90101 replied:
common이라는것은 진짜 디자인 컴포넌트 예를 들어 input, button같은거요.
이건 profile이라는 이름으로 하는게 좀더 의미있어 보입니다.

Monixc replied:
피드백 감사합니다! 컴포넌트의 역할과 구조에 대해 더 깊이 고민해볼 수 있는 계기가 되었습니다.

shadcn에서 가져오는 컴포넌트들(input, button 등)은 `/ui`, 그 외 재사용 가능한 컴포넌트들(`AppTabs` 등)은 `/common`에 위치시키는 기준으로 `ProfileSection`도 처음에는 `/common`에 배치했었습니다. `ProfileSection`이 `AppTabs`처럼 범용 컴포넌트라고 판단했기 때문인데, 멘토님 말씀을 듣고 보니 이 컴포넌트는 명확히 프로필 도메인에 종속적인 역할을 하네요.  `components/profile`로 구조를 수정해 반영했습니다!

다만 한 가지 여쭤보고 싶은 점이 있습니다.
현재 특정 페이지에 종속적인 컴포넌트는 `pages/(페이지)/_components`, 전역적으로 사용하는 도메인 컴포넌트는 `components/(도메인)` 형태로 관리하고 있습니다. 이렇게 **페이지 단위 + 도메인 단위**로까지 나누다 보니, 역할은 분명해졌지만 구조가 너무 파편화되는 건 아닐까 하는 고민이 생깁니다.
폴더가 여러 곳에 흩어지면서 오히려 구조의 집중도가 떨어지는 느낌도 있는 것 같아요...

* 현재 구조를 유지
* 아니면 `_components`로 분리해 놓은 컴포넌트들도 전부 `src/components/` 하위로 통합해 관리

어떤 방식이 더 나을지 멘토님의 의견이 궁금합니다!

ksy90101 commented on src/pages/developer/index.tsx:
```diff
@@ -0,0 +1,36 @@
+import Posts from "./_components/Posts.tsx";
+import Profile from "./_components/Profile.tsx";
+
+export const DeveloperPage = () => {
+  const [activeTab, setActiveTab] = useState<DeveloperTabValue>("profile");
+
+  const handleTabChange = (value: string) => {
+    if (DEVELOPER_TABS.some((tab) => tab.value === value)) {
+      setActiveTab(value as DeveloperTabValue);
+    }
+  };
```
as를 쓰지 않을 방법을 고민해보시면 좋겠습니다.

Monixc replied:
`AppTabs` 컴포넌트의 `onValueChange` prop을 `(value:string)=>void` 형태로 정의해서서, 실제로는 `profile|posts`처럼 제한된 값만 허용되어야 하는데도, `string` 전체를 허용하고 있었습니다.


`AppTabs`가 재사용되는 컴포넌트이고, 사용하는 페이지마다 탭의 타입이 달라, 다양한 문자열 리터럴을 받아야 해서 string으로 뭉뚱그려서 처리했었습니다. 이런 구조여서 사용하는 쪽에서는 `.some()`으로 런타임 검증을 하고, `value as DeveloperTabValue` 타입 단언하여 오류를 회피하는 방향을 선택했는데, 임시방편 코드였던 것 같습니다.


`AppTabs` 컴포넌트를 제네릭으로 개선해서 타입 단언을 사용하지 않는 방향으로 수정해보겠습니다.

Monixc committed:
fix: AppTabs 및 DeveloperPage 타입 안전성 강화

- AppTabsProps에 제네릭 타입 추가로 탭 값의 타입 안전성 향상
- DeveloperPage에서 handleTabChange 함수의 매개변수 타입을 DeveloperTabValue로 변경하여 타입 안전성 강화

Monixc reviewed (commented):
- AppTabsProps에 제네릭 타입 추가
- DeveloperPage에서 handleTabChange 함수의 매개변수 타입을 DeveloperTabValue로 변경

Monixc committed:
feat: ProfileSection 컴포넌트 추가 및 경로 수정

woo-dev-log reviewed (approved):
👍
